whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/SourceDataScan.java

```diff
@@ -117,28 +118,43 @@ private void generateReport(Map<String, List<FieldInfo>> tableToFieldInfos, Stri
 		SXSSFWorkbook workbook = new SXSSFWorkbook(100); // keep 100 rows in memory, exceeding rows will be flushed to disk
 
 		// Create overview sheet
-		Sheet sheet = workbook.createSheet("Overview");
+		Sheet overviewSheet = workbook.createSheet("Overview");
 		if (!scanValues) {
-			addRow(sheet, "Table", "Field", "Type", "N rows");
+			addRow(overviewSheet, "Table", "Field", "Type", "N rows");
 			for (String table : tables) {
 				for (FieldInfo fieldInfo : tableToFieldInfos.get(table))
-					addRow(sheet, table, fieldInfo.name, fieldInfo.getTypeDescription(), Long.valueOf(fieldInfo.rowCount));
-				addRow(sheet, "");
+					addRow(overviewSheet, table, fieldInfo.name, fieldInfo.getTypeDescription(), Long.valueOf(fieldInfo.rowCount));
+				addRow(overviewSheet, "");
 			}
 		} else {
-			addRow(sheet, "Table", "Field", "Type", "Max length", "N rows", "N rows checked", "Fraction empty");
-			for (String table : tables) {
-				for (FieldInfo fieldInfo : tableToFieldInfos.get(table))
-					addRow(sheet, table, fieldInfo.name, fieldInfo.getTypeDescription(), Integer.valueOf(fieldInfo.maxLength), Long.valueOf(fieldInfo.rowCount),
+			List<Sheet> valueSheets = new ArrayList<>();
+
+			addRow(overviewSheet, "Table", "Field", "Type", "Max length", "N rows", "N rows checked", "Fraction empty");
+			int sheetIndex = 0;
+			Map<String, String> sheetNameLookup = new HashMap<>();
+			for (String tableName : tables) {
+				// Create unique sheetNames from table name and index.
+				String sheetName = Table.createSheetNameFromTableName(tableName, sheetIndex);
+				sheetNameLookup.put(tableName, sheetName);
+
+				String tableNameIndexed = tableName;
+				if (tableName.length() > 31) {
+					tableNameIndexed = Integer.toString(sheetIndex) + '_' + tableName;

```

Comment:
Is this reversal compared to `createShssetNameFromTableName` intentional?

Reply:
Never mind, the index + sheet name order is the same.

Reply:
Just separated this a bit better from the main code.